flintrock/flintrock.py

```diff
@@ -297,6 +305,21 @@ def launch(
         raise UnsupportedProviderError(provider)
 
 
+def get_last_commit_sha(git_repository: str):
+    """
+    Retrieve the last commit's SHA from a repository hosted on Github.
+    """
+    parsed_url = urllib.parse.urlparse(git_repository)
+    if parsed_url.netloc != "github.com":
+        raise UsageError('Error: Retrieving the last commit\'s SHA only works with '
+                         'repositories hosted on Github.')
+    owner_repo = parsed_url.path[:-4] if parsed_url.path.endswith(".git") else parsed_url.path
+    url = "https://api.github.com/repos{owner_repo}/commits".format(owner_repo=owner_repo)
+    with urllib.request.urlopen(url) as response:
+        result = json.loads(response.read().decode('utf-8'))
+        return result[0]['sha']
```

Comment:
Hmm, I think this is mostly fine as-is since with "unexpected" failures like this we simply want the error to bubble up to the user so they can debug.

The only problem I see here is that a misleading index or lookup error (from the `[0]['sha']`) may conceal the underlying problem with getting the commit (e.g. the request "succeeds" but returns bad info). Not sure if there is a simple way to separate those two problems.